pkg/network/ovn_kubernetes.go

```diff
@@ -1258,16 +1258,19 @@ const (
 type targetZoneModeType struct {
 	// zoneMode indicates the target zone mode that CNO is supposed to converge to.
 	zoneMode InterConnectZoneMode
-	// "temporary", if true, indicates that the target zone mode is only temporary;
-	// it is used along with zoneMode=singlezone in order to temporarily switch to single-zone mode when upgrading
-	// from versions with no interconnect support (<=4.13). It has no effect if zoneMode=multizone.
-	temporary bool
```

Comment:
so this variable is now removed in 4.14.7 or whatever z stream this lands in right?
how will the new upgrades from nonIC to 4.14.z where this change lands look like? I see
`remove the temporary field from the configmap, since the ongoing-upgrade field already tracks that CNO is going through an upgrade` in the PR message so that means we no longer need to know where the phase is temporary or not, just need to know if upgrade is ongoing or not? why did we need the `temporary` flag originally? seems like it was redundant with ongoingUpgrade?

I guess my TL;DR question is is there a difference between going from 4.13.0 to 4.14.0 and 4.13.0 to 4.14.z where this lands functionally for the end user if this field is going away? Do we need to make updates to the original enhancement?

Reply:
I could have simplified the logic earlier on, but I wanted to leave a window open in case we were determined to allow single zone... 

Here's how it went.

The `temporary` field was how I envisaged to distinguish between being permanently in single zone (temporary=false) or just momentarily during the upgrade to 4.14 (temporary=true). The idea was that a configmap would track an upgrade to IC or a voluntary switch to single zone.

Then I needed another flag in the configmap, `ongoing-upgrade`, so that the CNO status manager could know whether an upgrade to IC was ongoing: it would only report version=4.14 when the upgrade was done, which was signaled by `ongoingUpgrade` getting removed from the configmap.

Well after 4.14 GA, we decided not to enable switching back to single zone, so now we can finally get rid of `temporary`.